51-android.rules

```diff
@@ -458,31 +459,27 @@ ATTR{idProduct}=="2d66", SYMLINK+="android_adb"
 ATTR{idProduct}=="428c", SYMLINK+="android_adb"
 #   Droid
 ATTR{idProduct}=="41db", SYMLINK+="android_adb"
-#   Xoom ID 1
-ATTR{idProduct}=="70a8", GOTO="adbfast"
-#   Xoom ID 2
-ATTR{idProduct}=="70a9", GOTO="adbfast"
+#   Xoom (70a8=mtp 70a9=mtp,adb)
+ATTR{idProduct}=="70a8", GOTO="mtp"
+ATTR{idProduct}=="70a9", GOTO="adbmtp"
+#   XT890/907/Razr (710d=mtp 710e=mtp,adb)
+ATTR{idProduct}=="710e", GOTO="adbmtp"
 #   Razr XT912
 ATTR{idProduct}=="4362", GOTO="adbfast"
-#   Moto XT1052
-#ATTR{idProduct}=="2e83", GOTO="adbfast"
-#   Moto E/G
-#ATTR{idProduct}=="2e76", GOTO="adbfast"
-#   Moto E/G (Dual SIM)
-#ATTR{idProduct}=="2e80", GOTO="adbfast"
-#   Moto E/G (Global GSM)
-#ATTR{idProduct}=="2e82", GOTO="adbfast"
-#   Moto x4
-#ATTR{idProduct}=="2e81", GOTO="adbfast"
 #   Droid Turbo 2
 ATTR{idProduct}=="2ea4", GOTO="adbfast"
+#   Atrix/Razr HD (2e32=mtp 2e33=mtp,adb)
+#   Razr M (2e50=mtp 2e51=mtp,adb)
+#   Moto G3 (2e76=mtp,adb 2e81=charge,adb 2e82=mtp 2e83=ptp 2e84=ptp,adb 2e24=rndis 2e25=rndis,adb)
 #   Moto Z3 Play, beckham, XT1929
 #     For details see: <https://github.com/M0Rf30/android-udev-rules/issues/264>
 #     18d1:4ee8=midi; 18d1:4ee9=midi+adb
 #     BP TOOLS: 2ee5=charging, mtp, ptp; 2ee6=charging+adb, mtp+adb, ptp+adb; 2ee7=rndis; 2ee8=rndis+adb; 18d1:4ee8=midi; 18d1:4ee9=midi+adb
 #     QCOM: 05c6:9091=charing+adb, mtp+adb, ptp+adb; 05c6:9092=charging, mtp, ptp; 18d1:4ee8=midi; 18d1:4ee9=midi+adb; 22b8:2e24=rndis; 22b8:2e25=rndis+adb
 ATTR{idProduct}=="2e24", GOTO="rndis"
 ATTR{idProduct}=="2e25", GOTO="adbrndis"
+ATTR{idProduct}=="2e33", GOTO="adbmtp"
+ATTR{idProduct}=="2e51", GOTO="adbmtp"
```

Comment:
I'm not sure that's feasible but could we try to keep individual devices separated? Ideally each device would be in its own section that fully describes what usb ids (and in which modes) it can show up as.

Of course since manufacturers do use the same IDs for different devices, we will have to somehow accommodate that, so here's an idea:

For Moto Z3 Play and Galaxy A4 (2016) tried keeping the following format:
```
#   <device name>, <codename>, <model #>
#     <any additional information, like links to a full investigation>
#     <ids that already appear elsewhere>
#     <ids that we can't yet reasonably use in code>
<rule>
<rule> # also <another mode that features the same id, but we don't have a way to express in code or distinguish>
<rule> # <explanation if the rule doesn't exactly match the mode the device is in>
```
The first two lines are pretty self-explanatory.
The next tow lines are in this format:
```
```